### PR TITLE
Update collections

### PR DESCRIPTION
**Pull request by tomnelson**
This PR updates our use of commons-collections to version4.2 and applies generics to (mostly) the Transformer implementations.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/transformer/FacetedTransformer.java [hunk 1]
```diff
@@ -29,31 +34,37 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.log4j.Logger;
 
-public class FacetedTransformer extends DocumentTransformer {
+public class FacetedTransformer extends BaseQueryLogicTransformer<Entry<Key,Value>,FacetsBase> {
     
     private static final Logger log = Logger.getLogger(FacetedTransformer.class);
     
+    private DocumentTransformer documentTransformer;
+
```
Comment (tomnelson) [inline warehouse/query-core/src/main/java/datawave/query/transformer/FacetedTransformer.java]: FacetedTransformer has the most disruptive changes. It no longer extends DocumentTransformer but instead creates and uses a DocumentTransformer as a delegate, then does its own transform with appropriate input and output types.

File: warehouse/query-core/src/main/java/datawave/query/tables/facets/MergedReadAhead.java [hunk 2]
```diff
@@ -4,11 +4,12 @@
 import java.io.IOException;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Queue;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-import org.apache.commons.collections.Buffer;
-import org.apache.commons.collections.BufferUtils;
-import org.apache.commons.collections.buffer.CircularFifoBuffer;
+//import org.apache.commons.collections4.Buffer;
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/tables/facets/MergedReadAhead.java]: Cleanup?

File: warehouse/query-core/src/main/java/datawave/query/transformer/FacetedTransformer.java [hunk 3]
```diff
@@ -113,9 +124,11 @@ public FacetedTransformer(String tableName, Query settings, MarkingFunctions mar
                     value.setDelegate(v.toString());
                     value.setNormalizedValue(v.toString());
                     
-                    FieldCardinalityBase fc = this.responseObjectFactory.getFieldCardinality();
-                    fc.setMarkings(markingFunctions.translateFromColumnVisibilityForAuths(attr.getColumnVisibility(), auths)); // reduces colvis based on
-                                                                                                                               // visibility
+                    FieldCardinalityBase fc = documentTransformer.getResponseObjectFactory().getFieldCardinality();
+                    fc.setMarkings(markingFunctions.translateFromColumnVisibilityForAuths(attr.getColumnVisibility(), documentTransformer.getAuths())); // reduces
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/transformer/FacetedTransformer.java]: Haha, looks like the formatter had fun with this.

File: web-services/client/src/main/java/datawave/webservice/query/result/event/Fielded.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package datawave.webservice.query.result.event;
+
+import java.util.List;
+
+public interface Fielded extends HasMarkings {
```
Comment (jwomeara) [inline web-services/client/src/main/java/datawave/webservice/query/result/event/Fielded.java]: Is this used anywhere?

**Comment by tomnelson** [comment]
closing and replacing with update_collections2 post merge of unique function changes

**Pull request closed without merging**